packages/types/src/interfaces/nft/enhanced-token-id.ts

```diff
@@ -0,0 +1,71 @@
+import { Struct } from '@polkadot/types';
+import { Codec, Registry } from '@polkadot/types/types';
+import { u8aToHex } from '@polkadot/util';
+import { CollectionId, SerialNumber, SeriesId } from './types';
+
+interface TokenIdValue {
+    collectionId: string | number,
+    seriesId: string | number,
+    serialNumber: string | number,
+}
+
+export class EnhancedTokenId extends Struct implements Codec {
+    constructor(registry: Registry, value?: TokenIdValue | [number, number, number] | any) {
+      super(
+        registry,
+        {
+            collectionId: 'u32',
```

Comment:
may be we just have them of the type
collectionId: CollectionId
seriesId: SeriesId
serialNumber: SerialNumber from ./types